### PR TITLE
Product add to cart

### PR DESCRIPTION
**Pull request by uestczmn**
augment AddToCart component, including some adjustments on size-selector, quantity-selector and addToCart button

### CODE REVIEW

**Review by jacinthechong — APPROVED** [review]

File: client/src/product-components/AddToCart/AddToCart.jsx [hunk 1]
```diff
@@ -1,14 +1,15 @@
 import React from "react";
+const axios = require('axios');
 
 class AddToCart extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      sizeValue: '0',
+      sizeValue: 'SelectSize',
```
Comment (LesliePajuelo) [inline client/src/product-components/AddToCart/AddToCart.jsx:8]: Is this intended to be read by the end user? If it's not specified in the BRD then I suggest just "Select size". Spacing and caps which look good to end users

File: client/src/product-components/AddToCart/AddToCart.jsx [hunk 2]
```diff
@@ -47,39 +57,53 @@ class AddToCart extends React.Component {
     });
   }
 
+  handleAddToCart() {
+    axios.post('/cart', {
+      sku_id: this.props.sizeList[this.state.selectedIndex].sku,
```
Comment (LesliePajuelo) [inline client/src/product-components/AddToCart/AddToCart.jsx:62]: This is a good candidate for optional chaining

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

File: client/src/product-components/AddToCart/AddToCart.jsx [hunk 3]
```diff
@@ -47,39 +57,53 @@ class AddToCart extends React.Component {
...
+    })
+    .then(function (response) {
```
Comment (LesliePajuelo) [inline client/src/product-components/AddToCart/AddToCart.jsx:64]: When it succeeds console log, when it fails console log? 

If you have to do this then at least a .finally

File: client/src/product-components/AddToCart/AddToCart.jsx [hunk 4]
```diff
@@ -47,39 +57,53 @@ class AddToCart extends React.Component {
...
-    // console.log(this.props.sizeList, '================this.props.sizeList');
+    // console.log(this.props.sizeList, this.state.selectedIndex, '================this.props.sizeList');
     //console.log(this.props.sizeList, this.state.selectedIndex, 'in render');
     return (
       <div>
-        <div className="SizeSelector">
+        <div  className="SizeSelector">
           Size:
-          <select value={this.state.sizeValue} onChange={this.handleSizeChange.bind(this)}>
+          <select value={this.state.sizeValue} disabled={this.props.sizeOutOfStock} onChange={this.handleSizeChange.bind(this)}>
```
Comment (LesliePajuelo) [inline client/src/product-components/AddToCart/AddToCart.jsx:81]: .bind creates and a returns a new function. This is def premature optimization, but to be frugal bind in the constructor and call in the render

File: client/src/product-components/AddToCart/AddToCart.jsx [hunk 5]
```diff
@@ -47,39 +57,53 @@ class AddToCart extends React.Component {
...
+            <option value={'SelectSize'}>{this.props.sizeOutOfStock ? 'OUT OF STOCK' : 'Select Size'}</option>
```
Comment (LesliePajuelo) [inline client/src/product-components/AddToCart/AddToCart.jsx:82]: "Select Size" should definitely be a variable at this point, 3rd time I've seen it and this usage is slightly different than the first two.

File: client/src/product-components/AddToCart/AddToCart.jsx [hunk 6]
```diff
@@ -47,39 +57,53 @@ class AddToCart extends React.Component {
...
             {
               this.props.sizeList.map((item, index) => {
```
Comment (LesliePajuelo) [inline client/src/product-components/AddToCart/AddToCart.jsx:84]: optional chaining when invoking functions in nested properties. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

File: client/src/product-components/AddToCart/AddToCart.jsx [hunk 7]
```diff
@@ -47,39 +57,53 @@ class AddToCart extends React.Component {
...
-                <option key={index} value={item.size}>{item.size}</option>
-              )
+                return (
+                  <option key={index} value={item.size}>{item.size}</option>
+                )
             })}
           </select>
         </div><br/>
 
         <div className="QuantitySelector">
```
Comment (LesliePajuelo) [inline client/src/product-components/AddToCart/AddToCart.jsx:92]: CSS class names, by convenction are snake case. All lower case with hyphens. so, `quantity-selector`

File: client/src/product-components/AddToCart/AddToCart.jsx [hunk 8]
```diff
@@ -47,39 +57,53 @@ class AddToCart extends React.Component {
...
           Quantity:
           <select value={this.state.quantityValue} onChange={this.handleQuantityChange.bind(this)}>
-            {
+            {this.state.sizeValue === 'SelectSize' ?  (<option value={'-'}> - </option>) :
               this.QuantityMaxList(this.props.sizeList[this.state.selectedIndex].quantity).map((quantity, index) => {
```
Comment (LesliePajuelo) [inline client/src/product-components/AddToCart/AddToCart.jsx:96]: By convention only classes are upper-case. The exception are functional components which are also capitalized when imported.

File: client/src/product-components/Overview.jsx [hunk 9]
```diff
@@ -41,12 +42,25 @@ class Overview extends React.Component {
...
     // console.log(sizeObj, sizeList, '=============sizeList');
     return sizeList;
   }
 
+  getSizeStock(sizeObj) {
+    var stock = 0;
+    for (var i in sizeObj) {
+      stock += sizeObj[i].quantity;
+    }
+    if (stock <= 0) {
```
Comment (LesliePajuelo) [inline client/src/product-components/Overview.jsx:58]: This is a rare case where a ternary really would be best

stock <= 0 ? true : false;

**Review by LesliePajuelo — COMMENTED** [review]
Good! No required changes. Before you finish the project it'd be worth removing all the commented out code.

**Pull request merged**